soc/nxp/mcx/mcxa/soc.c

```diff
@@ -17,11 +17,7 @@
 #include <zephyr/init.h>
 #include <soc.h>
 
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
```

Comment:
I believe the `ifdef` here is not useless- we don't compile in the code implementing `SystemInit` unless `CONFIG_PLATFORM_SPECIFIC_INIT` is set: https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/hal_nxp.cmake#L101

So we might need a change to the HAL as well there, to use `CONFIG_SOC_RESET_HOOK`?

Reply:
ok, re-added it to be consistent with other usages, can be removed later everywhere if we decide it is not needed.

Reply:
Doesn’t the soc always select the option?